components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApplicationsApiServiceImpl.java

```diff
@@ -1280,11 +1284,26 @@ public Response applicationsApplicationIdOauthKeysKeyMappingIdPut(String applica
             throws APIManagementException {
 
         String username = RestApiCommonUtil.getLoggedInUsername();
-            APIConsumer apiConsumer = APIManagerFactory.getInstance().getAPIConsumer(username);
-            Application application = apiConsumer.getApplicationByUUID(applicationId);
-            if (application != null) {
-                ApplicationKeyDTO appKey = getApplicationKeyByAppIDAndKeyMapping(applicationId, keyMappingId);
-                if (RestAPIStoreUtils.isUserOwnerOfApplication(application) && appKey != null) {
+        APIConsumer apiConsumer = APIManagerFactory.getInstance().getAPIConsumer(username);
```

Comment:
Just doublecheck whether user need to send the keyManager UUID to this call.
Ideally things needs to be consistent and need to pass keyManagerName instead similar to other operations.